cid.go

```diff
@@ -825,3 +826,35 @@ func CidFromReader(r io.Reader) (int, Cid, error) {
 
 	return len(br.dst), Cid{string(br.dst)}, nil
 }
+
+// IsIdentity checks whether a CID has multihash.IDENTITY code.
+// Note that this function is optimized to return fast.
+// Checking for multihash.IDENTITY code in this function does not fully verify the validity of the CID.
+func (c Cid) IsIdentity() (bool, error) {
+
+	if c.Version() == 0 {
+		return false, nil
+	}
+
+	var n int
+	// Skip version.
+	_, n, err := uvarint(c.str[n:])
+	if err != nil {
+		return false, err
+	}
+	n += n
```

Comment:
Did you mean do declare a `bytesRead int`, and add `n` every time you read a varint? Now you're just overwriting `n` every time.